nixos/doc/manual/release-notes/rl-2405.section.md

```diff
@@ -8,6 +8,11 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 <!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->
 
+- Users now default to a primary group named the same as the user, and with group id same as user id.
+  Before it was "users" with group id 100 for normal users and "" (empty) for system users -- which meant system users required explicitly setting the primary group.
+  This change gives better control over permissions/file sharing and aligns NixOS with other distros like Arch Linux/Debian/Fedora/Ubuntu.
+  The new functionality is gated behind an option called `users.solitaryByDefault`, which defaults to true for `system.stateVersion >= 24.05`.
```

Comment:
So, for any user that created their system using unstable *after* the branch-off of 23.11, won't this break things? Such users will have `stateVersion = "24.05";` but their files will all be owned by `me:users`.

Reply:
Yes, good point. (Although the breakage does seem "soft" to me -- all my $HOME files will still be accessible even without "users" membership.)

Should we bump the stateVersion in this PR one more release into the future (24.11)?

Or any other ideas?

(I was considering keeping it entirely opt-in for now, but I think we should have a plan to have solitaryByDefault = true _eventually_.)

There's also https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574 -- would posting there be enough?

Reply:
The latest version of this PR has per-user groups enabled behind stateVersion>=24.11, so unstable users won't get their file permissions changed.